ajax_with_jquery_exercise/index.js

```diff
@@ -0,0 +1,271 @@
+		} else {
+			loggedIn = false;
+			localStorage.setItem("hos_loggedIn", loggedIn);
+			toggleLoginMenuMode($lnkLogin, $lnkSignup, loggedIn);
+			setChecksVisible($list, loggedIn);
+		}
+	});
+
+	$lnkSignup.on('click', function() {
+		$signupDiv.eq(0).slideToggle(250);
+	});
+
+	// Combine these two into a single function and pass whether login or signup?
+	$btnLogin.on('click', function(event) {
+		var username = $("#input_text3").val();
+		var password = $("#input_text4").val();
+		
+		if ($(event.target).attr("id") === "login_submit") {
+			$.ajax({
+			    method: "POST",
+			    headers: {
+			        "Content-Type": "application/json"
+			      },
+			    url: "https://hn-favorites.herokuapp.com/login",
+			    data: JSON.stringify({
+			        email: username, 
+			        password: password
+			    })
+			})
+			.then(function(data) { 
+				loginMaintenance(data);
+			})
+			.fail(function(err) {
+				console.warn("Auth Token Error: " + err);
+				alert("Log In: Please enter a valid email address and password!");
+			});
+		}
+	});
+
+	$btnSignup.on('click', function(event) {
+		var username = $("#input_text3").val();
+		var password = $("#input_text4").val();
+		
+		if ($(event.target).attr("id") === "signup_submit") {
+			$.ajax({
+			    method: "POST",
+			    headers: {
+			        "Content-Type": "application/json"
+			      },
+			    url: "https://hn-favorites.herokuapp.com/signup",
+			    data: JSON.stringify({
+			        email: username, 
+			        password: password
+			    })
+			})
+			.then(function(data) { 
+				loginMaintenance(data);
+			})
+			.fail(function(err) {
+				console.warn("Auth Token Error: " + err);
+				alert("Sign Up: Please enter a valid email address and password!");
+			});
+		}
+	});
+
+	function loginMaintenance(data) {
+		console.log(data.auth_token);
+		localStorage.setItem("hos_autoToken", data.auth_token);
+		localStorage.setItem("hos_loggedIn", true);
+		loggedIn = true;
+		$loginDiv.eq(0).slideUp(250);
+		$signupDiv.eq(0).slideUp(250);
+		toggleLoginMenuMode($lnkLogin, $lnkSignup, loggedIn);
+		setChecksVisible($list, loggedIn);
+	}
+
+	$list.on('click', function(event){
+		if ($(event.target).attr('type') === 'checkbox') {
+			if ($(event.target).is(':checked')) {
+				
+				var $siblings = $(event.target).siblings(); // Get Parent
+				var $key = $(event.target).attr('id');
+				// Add this checkbox to the local container of favorites.
+				favContainer[$key] = {
+					$url: $($siblings[1]).text().replace("(", "").replace(")", ""),
+					$by: $($siblings[2]).text().slice(3),
+					$title: $($siblings[0]).text()
+				}
+
+				// FAIL
+				$.ajax({
+			    method: "POST",
+			    headers: {
+			        "Content-Type": "application/json",
+			        "Authorization": localStorage.getItem("hos_autoToken")
+			      },
+			    url: "https://hn-favorites.herokuapp.com/stories.json",
+			    data: {
+		        hacker_news_story: {						///// THIS IS FAILING. JSON Turns this into a bad string
```

Comment:
Not sure what's going on here, but you're right that the ajax is failing.  It's not sending any data in the http request.